### PR TITLE
Match container/storage adding extra parameter to PutLayer.

### PR DESCRIPTION
**Pull request by adelton**
This catches up to change in https://github.com/containers/storage/pull/156.

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@adelton need to sign this one.  Otherwise the change LGTM.

**Comment by adelton** [comment]
Signed-off-by added -- sorry about that.

**Comment by adelton** [comment]
> You need to change the actual interface as well.

Could you elaborate? The https://github.com/containers/storage/pull/156 was not merged yet -- I assume that once it gets merged, it will start to be used automatically. Won't it?

File: storage/storage_image.go [hunk 1]
```diff
@@ -556,7 +556,8 @@ func (s *storageImageDestination) Commit() error {
 			return errors.Errorf("error applying blob %q: content not found", blob.Digest)
 		}
 		// Build the new layer using the diff, regardless of where it came from.
-		layer, _, err := s.imageRef.transport.store.PutLayer(id, lastLayer, nil, "", false, diff)
+		options := (*storage.LayerOptions)(nil)
+		layer, _, err := s.imageRef.transport.store.PutLayer(id, lastLayer, nil, "", false, options, diff)
```
Comment (mtrmac) [inline storage/storage_image.go]: Why not just `…, false, nil, diff)`?

Or if you wanted to make it easy to add fields in the future, `options := &storage.LayerOptions{}`.  Setting the typed `nil` is a surprising mix of the two ideas which is more complex than either of them.

**Comment by mtrmac** [comment]
> I assume that once it gets merged, it will start to be used automatically. Won't it?

It will in this package (`vendor.conf` specifies `master` branch from c/storage); but we will need an actual updating PR for `skopeo`, which also specifies `master` but actually does vendoring and keeps the vendored in the git repository; c/image runs `skopeo` tests for each PR.

I guess I can take care of that bit of procedure after it becomes necessary.

File: storage/storage_image.go (hunk 1 above)
Comment (adelton) [reply storage/storage_image.go]: I thought having the `nil` typed would make it more explicit in showing what we try to achieve.

I've changed it to untyped `nil` now -> 1efad7d255b3e1f0250beccbd8d3dc2608e02365.

**Comment by adelton** [comment]
> I guess I can take care of that bit of procedure after it becomes necessary.

Thank you.

This PR is waiting for https://github.com/containers/storage/pull/156 to be merged, so feel free to set any tags / anything about this one to put it aside until the parent one is resolved.

**Comment by mtrmac** [comment]
:+1: pending containers/storage#156 and tests. Thanks!

**Comment by mtrmac** [comment]
Ended up merging equivalent #436 instead.

**Pull request closed without merging**